### PR TITLE
Switch out use of S3Upload for S3UploadV2 throughout

### PR DESCRIPTION
**Pull request by sihil**
This addresses #291 by switching all uses over to the cleaner S3UploadV2 and deleting the unused task. All of the tests pass, but I think this still needs some care when rolling out - it impacts the `self-deploy` (i.e. Riff-Raff's deploy), `aws-s3`, `autoscaling`, `elasticsearch`.

(It also impacts the `uploadArtifacts` actions on the `executable-jar-webapp`, `native-packager-webapp` and `jetty-webapp` deployment types - although I very much doubt that any of those are still in use).

### CODE REVIEW

File: magenta-lib/src/main/scala/magenta/deployment_type/S3.scala [hunk 1]
```diff
@@ -106,18 +107,18 @@ object S3 extends DeploymentType with S3AclParams {
         assert(data.isDefined, s"Cannot find resource value for ${bucketResource(pkg)} (${pkg.apps.head} in ${parameters.stage.name})")
         data.get.value
       }
+      val prefix:String = S3UploadV2.prefixGenerator(
+        stack = if (prefixStack(pkg)) Some(stack) else None,
+        stage = if (prefixStage(pkg)) Some(parameters.stage) else None,
+        packageName = if (prefixPackage(pkg)) Some(pkg.name) else None
+      )
```
Comment (jennysivapalan) [inline magenta-lib/src/main/scala/magenta/deployment_type/S3.scala]: This looks similar to the logic in `Autoscaling.scala`, is it worth `S3UploadV2` having a method that has `pkg` as the parameter and return the result of `prefixGenerator`?

Comment (sihil) [inline magenta-lib/src/main/scala/magenta/deployment_type/S3.scala]: It should be identical. The problem is that it also needs the parameters from the class, so it's quite difficult to dry out without passing in seven parameters. I could extract it into a param trait.

Comment (sihil) [inline magenta-lib/src/main/scala/magenta/deployment_type/S3.scala]: After more thought I've decided not to bother DRYing this out. It is minimal code duplication.

Comment (jennysivapalan) [inline magenta-lib/src/main/scala/magenta/deployment_type/S3.scala]: Agreed - don't want to make you do more work for a few lines!

**Comment by sihil** [comment]
I've tested this locally with autoscaling deploys and will test the self deploy as soon as I've pushed out this PR.

Any thoughts @philwills

**Comment by philwills** [comment]
Any reason to keep the V2 suffix? Looks like this could just be the new S3Upload.

**Comment by sihil** [comment]
Nope - just renamed it.

**Comment by philwills** [comment]
👍 👍 👍

**Pull request merged**